crates/sui-metrics/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::OnceCell;
+use prometheus::{register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+
+pub use scopeguard;
+
+#[derive(Debug)]
+pub struct Metrics {
+    pub tasks: IntGaugeVec,
+    pub futures: IntGaugeVec,
+}
+
+impl Metrics {
+    fn new(registry: &Registry) -> Self {
+        Self {
+            tasks: register_int_gauge_vec_with_registry!(
+                "monitored_tasks",
+                "Number of running tasks per callsite.",
+                &["callsite"],
+                registry,
+            )
+            .unwrap(),
+            futures: register_int_gauge_vec_with_registry!(
+                "monitored_futures",
+                "Number of pending futures per callsite.",
+                &["callsite"],
+                registry,
+            )
+            .unwrap(),
+        }
+    }
+}
+
+static METRICS: OnceCell<Metrics> = OnceCell::new();
+
+pub fn init_metrics(registry: &Registry) {
+    METRICS
+        .set(Metrics::new(registry))
+        .expect("sui_metrics::init_metrics duplicate init")
+}
```

Comment:
Does this expect fail in our tests which spawn multiple nodes in the same process?

Reply:
yes, fixed.